src/components/common/tooltips/Tooltip.js

```diff
@@ -0,0 +1,190 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import React, { PropTypes } from 'react';
+import _ from 'lodash';
+
+import styles from './Tooltip.css';
+
+class Tooltip extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { offset: { top: 0, left: 0 } };
+  }
+
+  componentDidMount() {
+    this.calculateOffset();
+  }
+
+  calculateOffset() {
+    const { offset: propOffset, tail } = this.props;
+    const offset = {};
+    const tooltipRect = this.element.getBoundingClientRect();
+    if (tail) {
+      const tailRect = this.tailElem.getBoundingClientRect();
+      const tailCenter = {
+        top: tailRect.top + (tailRect.height / 2),
+        left: tailRect.left + (tailRect.width / 2),
+      };
+      offset.top = -tailCenter.top + tooltipRect.top + propOffset.top;
+      offset.left = -tailCenter.left + tooltipRect.left + propOffset.left;
+    } else {
+      let leftOffset;
+      let topOffset;
+      switch (this.props.side) {
```

Comment:
Could you add some tests around this logic? I think it'd be useful to prevent future regressions to know what the offsets should be (and also as pseudo-documentation/examples of how the offsets work).